Assets/MixedRealityToolkit-Tests/Core/TestFixture_01_MixedRealityToolkitTests.cs

```diff
@@ -0,0 +1,543 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Core.Interfaces;
+using Microsoft.MixedReality.Toolkit.Core.Services;
+using Microsoft.MixedReality.Toolkit.Tests.Services;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace Microsoft.MixedReality.Toolkit.Tests.Core
+{
+    public class TestFixture_01_MixedRealityToolkitTests
+    {
+        #region Service Locator Tests
+
+        [Test]
+        public void Test_01_InitializeMixedRealityToolkit()
+        {
+            TestUtilities.CleanupScene();
+
+            TestUtilities.InitializeMixedRealityToolkit();
+            GameObject gameObject = GameObject.Find(nameof(MixedRealityToolkit));
+            Assert.AreEqual(nameof(MixedRealityToolkit), gameObject.name);
+        }
+
+        [Test]
+        public void Test_02_TestNoMixedRealityConfigurationFound()
+        {
+            LogAssert.Expect(LogType.Error, "No Mixed Reality Configuration Profile found, cannot initialize the Mixed Reality Toolkit");
+
+            MixedRealityToolkit.Instance.ActiveProfile = null;
+
+            Assert.IsFalse(MixedRealityToolkit.HasActiveProfile);
+            Assert.IsNull(MixedRealityToolkit.Instance.ActiveProfile);
+
+            TestUtilities.CleanupScene();
+        }
+
+        [Test]
+        public void Test_03_CreateMixedRealityToolkit()
+        {
+            TestUtilities.InitializeMixedRealityToolkitScene();
+
+            // Create Test Configuration
+            Assert.AreEqual(0, MixedRealityToolkit.ActiveSystems.Count);
+            Assert.AreEqual(0, MixedRealityToolkit.RegisteredMixedRealityServices.Count);
+        }
+
+        #endregion Service Locator Tests
+
+        #region IMixedRealityDataprovider Tests
+
+        [Test]
+        public void Test_04_01_RegisterMixedRealityDataProvider()
+        {
+            TestUtilities.InitializeMixedRealityToolkitScene();
+
+            // Register
+            MixedRealityToolkit.Instance.RegisterService(typeof(ITestDataProvider1), new TestDataProvider1("Test Data Provider 1", 10));
+
+            // Retrieve
+            var extensionService1 = MixedRealityToolkit.Instance.GetService(typeof(ITestDataProvider1));
+
+            // Tests
+            Assert.IsEmpty(MixedRealityToolkit.ActiveSystems);
+            Assert.AreEqual(1, MixedRealityToolkit.RegisteredMixedRealityServices.Count);
+
+            // Tests
+            Assert.IsNotNull(extensionService1);
+        }
+
+        [Test]
+        public void Test_04_02_01_UnregisterMixedRealityDataProviderByType()
+        {
+            // Redo register test service for a valid instance.
+            Test_04_01_RegisterMixedRealityDataProvider();
```

Comment:
Rather than re-running a test (which may fail) please consider a helper function that sets the desired initial conditions?

Reply:
Why when the state is satisfactory from the previous test? 

If the previous test would fail then this one should too.

Reply:
In my previous test dev life, I found tests to be best (and most reliable) when they could be run completely in isolation. It is possible to train the system if one test depends upon another.

Reply:
Moving the code to a different location doesn't add any benefit, and adds more overhead to maintainability.

While I do agree that tests should run in isolation, I think that applies to the specific modules being tested.